mlrun/db/base.py

```diff
@@ -142,6 +143,91 @@ def delete_function(self, name: str, project: str = ""):
     def list_functions(self, name=None, project="", tag="", labels=None):
         pass
 
+    @abstractmethod
+    def tag_objects(
+        self,
+        project: str,
+        tag_name: str,
+        tag_objects: schemas.TagObjects,
+        replace: bool = False,
+    ):
+        pass
+
+    @abstractmethod
+    def delete_objects_tag(
+        self,
+        project: str,
+        tag_name: str,
+        tag_objects: schemas.TagObjects,
+    ):
+        pass
+
+    @abstractmethod
+    def tag_artifacts(
+        self,
+        artifacts,
+        project: str,
+        tag_name: str,
+        replace: bool = False,
+    ):
+        pass
+
+    @abstractmethod
+    def delete_artifacts_tags(
+        self,
+        artifacts,
+        project: str,
+        tag_name: str,
+    ):
+        pass
+
+    @staticmethod
+    def _resolve_artifacts_to_tag_objects(
+        artifacts,
+    ) -> schemas.TagObjects:
+        """
+        :param artifacts: Can be a list of :py:class:`~mlrun.artifacts.Artifact` objects or
+            dictionaries, or a single object.
+        :return: :py:class:`~mlrun.api.schemas.TagObjects`
+        """
+        # to avoid circular imports we import here
+        import mlrun.artifacts.base
+
+        if not isinstance(artifacts, list):
+            artifacts = [artifacts]
+
+        artifact_identifiers = []
+        for artifact in artifacts:
+            artifact_obj = (
+                artifact.to_dict()
+                if isinstance(artifact, mlrun.artifacts.base.Artifact)
+                else artifact
+            )
+            artifact_identifiers.append(
+                schemas.ArtifactIdentifier(
+                    key=mlrun.utils.get_in_artifact(artifact_obj, "key"),
+                    uid=mlrun.utils.get_in_artifact(artifact_obj, "uid"),
+                    kind=mlrun.utils.get_in_artifact(artifact_obj, "kind"),
+                    iter=mlrun.utils.get_in_artifact(artifact_obj, "iter"),
+                )
+            )
+        return schemas.TagObjects(kind="artifact", identifiers=artifact_identifiers)
+
+    @staticmethod
+    def _resolve_artifacts_identifiers_to_tag_objects(
```

Comment:
What is this function? Cannot see it used anywhere (and not sure I really understand its logic...)

Reply:
leftovers, removed